### PR TITLE
Zfits 2012 fix

### PR DESCRIPTION
**Pull request by mackaiver**
This contains a number of fixes that hopefully allow for reading all FACT related .fits and zfits files. 
I tested this on a small sample of real data and monte carlo files (even data from 2012. they have weird header strings.).  Also added soem unittests.
It also cleans up some of the stuff I found in the code of  Michael B.

### CODE REVIEW

File: src/main/java/fact/io/zfits/BlockHeader.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 		
 		private final short id;
 		// using int and casting to short saves us the work to write '(short)x' everywhere above
-		private Prozessor(int id) {
+		Prozessor(int id) {
```
Comment (MaxNoe) [inline src/main/java/fact/io/zfits/BlockHeader.java]: It's really small and nothing really important, but could we remove the german naming here?

Comment (mackaiver) [inline src/main/java/fact/io/zfits/BlockHeader.java]: Yeah. This stuff is everywhere. Didnt find all spots.

**Comment by jebuss** [comment]
works for me, accepted!

**Comment by jebuss** [comment]
+1

**Pull request merged**